cou/steps/plan.py

```diff
@@ -23,17 +23,25 @@
 logger = logging.getLogger(__name__)
 
 
-async def generate_plan(args: Analysis) -> UpgradeStep:
+async def generate_plan(analysis_result: Analysis) -> UpgradeStep:
     """Generate plan for upgrade.
 
     :param args: Analysis result.
     :type args: Analysis
     :return: Plan with all upgrade steps necessary based on the Analysis.
     :rtype: UpgradeStep
     """
-    logger.info(args)  # for placeholder
+    target = (
+        analysis_result.current_cloud_os_release.next_release
+        if analysis_result.current_cloud_os_release
+        else None
+    )
     plan = UpgradeStep(description="Top level plan", parallel=False, function=None)
     plan.add_step(
         UpgradeStep(description="backup mysql databases", parallel=False, function=backup)
     )
+    for app in analysis_result.apps:
+        app_upgrade_plan = app.generate_upgrade_plan(target)
+        if app_upgrade_plan:
+            plan.add_step(app_upgrade_plan)
```

Comment:
Right now we are not differentiating what is control plane from what is data plane. In a serial upgrade if we respect the `UPGRADE_ORDER` and every application know what it should do, then we are ok. However, this will be necessary on parallel upgrades.

In any case I've added a upgrade plan step and the structure is almost the same.